apps/emqx/src/emqx_router.erl

```diff
@@ -266,18 +266,15 @@ mria_batch_v1(Batch) ->
 batch_get_action(Op) ->
     element(1, Op).
 
--spec select(Spec, _Limit :: pos_integer(), Continuation) ->
-    {[emqx_types:route()], Continuation} | '$end_of_table'
-when
-    Spec :: {_TopicPat, _DestPat},
-    Continuation :: term() | '$end_of_table'.
-select(MatchSpec, Limit, Cont) ->
-    select(get_schema_vsn(), MatchSpec, Limit, Cont).
+-spec stream(_Spec :: {_TopicPat, _DestPat}) ->
```

Comment:
Note: this could potentially impact the performance of querying the topics, since from brief reading of the code streams create cons cells on every operation.

Reply:
Slight elaboration: they already do, stream-over-2-router-tables is already in use on master. This PR essentially just turns it into stream-over-4-router-tables (in the worst case).